### DISCUSSION TITLE
Split access between bitwarden and vaultwarden

### DISCUSSION THREAD

**Discussion by connorwilliams-aps** [dani-garcia/vaultwarden#2152] [Ideas]
Is there any was to implement a "hybrid" environment? To clarify. I have several users who use bitwarden personally and are already using the desktop apps, extensions etc from the official channel. I am not able to add these users to my org as they do not exist within it. I am trying to make it so they can keep their personal bitwarden accounts, and just provide access to my vaultwarden instance. Is this an upstream api limitation?

**Comment by BlackDex** [comment]
You can't interconnect two different (self-)hosted environments.
You also can't have two different accounts using one app/extention.
I think there is a feature request somewhere at Bitwarden regarding that because of people who have a personal account in the cloud, and there company has a self-hosted version, which is the same as you are mentioning, but i think there is nothing done regarding that request.

**Comment by jjlin** [comment]
Bitwarden was not designed with federation in mind, so it's pretty unlikely that there will ever be support for having entries from multiple Bitwarden/Vaultwarden instances consolidated into one client view.

However, upstream Bitwarden has started working on adding client support for switching between accounts (see https://github.com/bitwarden/jslib/pull/491). The feature isn't released yet, and they are currently only implementing it in the desktop client for some reason.

For the browser extension, I find that the most practical way to access multiple Bitwarden accounts is to just use different browser profiles. In general, having separate browser profiles for work and personal stuff is probably a good idea anyway.
